core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -174,11 +174,8 @@ class NewHadoopRDD[K, V](
             }
           }
         } catch {
-          case e: Exception => {
-            if (!Utils.inShutdown()) {
-              logWarning("Exception in RecordReader.close()", e)
-            }
-          }
+          case e: Exception => 
+            logWarning("Exception in RecordReader.close()", e)
```

Comment:
Ah sorry, I mean revert. Although I kind of like the additional follow-on changes, for this PR we should foremost focus on patching up the exception. If there's any question about any extra changes, like the one you raised, I'd favor just reverting that part.